app/routers/organizations.py

```diff
@@ -1,15 +1,43 @@
 from fastapi import APIRouter, status, HTTPException
 from database import client
 from models import Organization
+from fastapi.responses import JSONResponse
+import string
+import random
 
 router = APIRouter()
 
 
+@router.post("/create-org/{org_name}")
+async def create_organization(org_name: str):
+    if org_name is not None:
+        # create an API key
+        char_set = string.ascii_letters + string.punctuation
+        urand = random.SystemRandom()
+        org_key = "".join([urand.choice(char_set) for _ in range(20)])
+
+        # check if API key exists
+        if (client.quiz.organization.find_one({"org_key": org_key})) is None:
+            org_data = {"org_name": org_name, "org_key": org_key}
+            client.quiz.organization.insert_one(org_data)
+            return JSONResponse(status_code=status.HTTP_201_CREATED, content=org_name)
+
+        raise HTTPException(
+            status_code=500,
+            detail="Duplicate API key. Please try again.",
+        )
+
+
 @router.get("/check-auth-token/{api_key}", response_model=Organization)
 async def check_auth_token(api_key: str):
 
     if (
-        org := client.quiz.organization.find_one({"org_key": api_key}, {"org_name": 1})
+        org := client.quiz.organization.find_one(
+            {"org_key": api_key},
+            {
+                "org_name": 1,
```

Comment:
What is `1`? Why does it seem hardcoded?

Reply:
That's the syntax. Instead of returning all the data from the query, it returns only the requested fields.
https://www.mongodb.com/docs/manual/tutorial/project-fields-from-query-results/

Reply:
Ah okay cool 👌

Reply:
You won't need to do any of this in this way if you defined an appropriate OrganizationResponse model and passed that in the `response_model` argument in the decorator. This is how we have done this for all the other routes. So, let's stay consistent with that, okay?

Reply:
Yes yes, what @dalmia says sounds good